anvio/cazymes.py

```diff
@@ -0,0 +1,259 @@
+    def __init__(self, args, run=run, progress=progress):
+        """Setup a CAZyme database for anvi'o
+
+        http://www.cazy.org/
+
+        Parameters
+        ==========
+        args : argparse.Namespace
+            See `bin/anvi-setup-cazymes` for available arguments
+            - cazyme_data_dir : str, optional
+                The directory where the CAZyme data should be stored. If not provided, the data will be stored in the anvi'o data directory.
+            - reset : bool, optional
+                If True, the data directory will be deleted and recreated. Defaults to False.
+        run : terminal.Run, optional
+            An object for printing messages to the console.
+        progress : terminal.Progress, optional
+            An object for printing progress bars to the console.
+        """
+
+        self.args = args
+        self.run = run
+        self.progress = progress
+        self.cazyme_data_dir = args.cazyme_data_dir
+
+        filesnpaths.is_program_exists('hmmpress')
+        
+        if self.cazyme_data_dir and args.reset:
+            raise ConfigError("You are attempting to run CAZyme setup on a non-default data directory (%s) using the --reset flag. "
+                              "To avoid automatically deleting a directory that may be important to you, anvi'o refuses to reset "
+                              "directories that have been specified with --cazyme-data-dir. If you really want to get rid of this "
+                              "directory and regenerate it with CAZyme data inside, then please remove the directory yourself using "
+                              "a command like `rm -r %s`. We are sorry to make you go through this extra trouble, but it really is "
+                              "the safest way to handle things." % (self.cazyme_data_dir, self.cazyme_data_dir))
+
+        if not self.cazyme_data_dir:
+            self.cazyme_data_dir = os.path.join(os.path.dirname(anvio.__file__), 'data/misc/CAZyme')
+
+        filesnpaths.is_output_dir_writable(os.path.dirname(os.path.abspath(self.cazyme_data_dir)))
+
+        self.resolve_database_url()
+
+        if not args.reset and not anvio.DEBUG:
+            self.is_database_exists()
+
+        if args.reset:
+            filesnpaths.gen_output_directory(self.cazyme_data_dir, delete_if_exists=True, dont_warn=True)
+        else:
+            filesnpaths.gen_output_directory(self.cazyme_data_dir)
+
+    def resolve_database_url(self):
+        """Create path to CAZyme ftp
+
+        Added self values
+        ================= 
+        - self.page_index : string
+            version of CAZyme database
+
+        """
+        if self.args.cazyme_version:
+            self.page_index = self.args.cazyme_version 
+            self.run.info('Attempting to use version', self.args.cazyme_version)
+        else:
+            self.page_index = 'V11'
+            self.run.info_single('No CAZyme version specified. Using current release.')
```

Comment:
Is the current release always going to be 'V11'? :)

I think you should change this to 
```
self.run.info_single(f"No CAZyme version specified. Using {self.page_index}.")
```
to be more explicit in case we lag behind updating the default when CAZymes has an update.